src/extension.c

```diff
@@ -281,6 +282,20 @@ ts_extension_is_loaded(void)
 			 * that, for example, the catalog does not go looking for things
 			 * that aren't yet there.
 			 */
+			if (extstate == EXTENSION_STATE_TRANSITIONING)
+			{
+				/* when we are updating the extension, we execute
+				 * scripts in post_update.sql after setting up the
+				 * the dependencies. At this stage, TS
+				 * specific functionality is permitted as we now have
+				 * all catalogs and functions in place
+				 */
+				char *update_script_stage =
+					(char *) GetConfigOption("timescaledb.update_script_stage", true, false);
```

Comment:
Why casting a `const char *` to `char *`? This shouldn't be modified so change to:

```suggestion
				const char *update_script_stage =
					GetConfigOption("timescaledb.update_script_stage", true, false);
```

I would also `#define` the config variable name:

```
#define UPDATE_SCRIPT_STAGE_CONFIG_OPTION "timescaledb.update_script_stage"
```

Actually, even better: Move this to a function with a descriptive name:

```
static bool
is_in_post_update_stage(void)
{
   ....
}
```

Reply:
added #define.  Missed revert from char* to const char*. Modified.